google/cloud/internal/http_payload.cc

```diff
@@ -0,0 +1,60 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/internal/http_payload.h"
+#include "google/cloud/internal/curl_impl.h"
+#include "google/cloud/log.h"
+#include <sstream>
+
+namespace google {
+namespace cloud {
+namespace rest_internal {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+
+CurlHttpPayload::CurlHttpPayload(std::unique_ptr<CurlImpl> impl,
+                                 Options options)
+    : impl_(std::move(impl)), options_(std::move(options)) {}
+
+StatusOr<std::size_t> CurlHttpPayload::Read(absl::Span<char> buffer) {
+  return impl_->Read(buffer);
+}
+
+std::multimap<std::string, std::string> CurlHttpPayload::Trailers() const {
+  // TODO(sdhart): This currently returns all the previously available headers
+  // and any received trailers. Given that there should be a header listing
+  // what trailers to expect, this function could be improved such that it
+  // interrogates the "Trailer:" header and construct a map of only the
+  // trailers.
+  return impl_->headers();
+}
+
+StatusOr<std::string> ReadAll(std::unique_ptr<HttpPayload> payload,
```

Comment:
If you are trying to add placeholders or an scaffold to build around, this function seems premature.